coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -482,41 +537,138 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
+    //  the limits of max range of data exported
     validateSplitLeafPlan(logicalPlan)
     val qParams = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
     // get a mapping of assignments to time-ranges to query
+    val lookbackMs = getLookBackMillis(logicalPlan).max
+    val offsetMs = getOffsetMillis(logicalPlan).max
     val assignmentRanges = {
       // "distinct" in case this is a BinaryJoin
       val partitions = getPartitions(logicalPlan, qParams).distinct.sortBy(_.timeRange.startMs)
+      require(!partitions.isEmpty, s"Partition assignments is not expected to be empty for query ${qParams.promQl}")
       val timeRange = TimeRange(1000 * qParams.startSecs, 1000 * qParams.endSecs)
-      val lookbackMs = getLookBackMillis(logicalPlan).max
-      val offsetMs = getOffsetMillis(logicalPlan).max
       val stepMsOpt = if (qParams.startSecs == qParams.endSecs) None else Some(1000 * qParams.stepSecs)
       getAssignmentQueryRanges(partitions, timeRange,
         lookbackMs = lookbackMs, offsetMs = offsetMs, stepMsOpt = stepMsOpt)
     }
-    // materialize a plan for each range/assignment pair
-    val plans = assignmentRanges.map { case (part, range) =>
-      val newParams = qParams.copy(startSecs = range.startMs / 1000,
-                                   endSecs = range.endMs / 1000)
-      val newContext = qContext.copy(origQueryParams = newParams)
-      materializeForPartition(logicalPlan, part, newContext)
+    val execPlans = if (assignmentRanges.isEmpty) {
+      // Assignment ranges empty means we cant run this query fully on one partition and needs
+      // remote raw export Check if the total time of raw export is within the limits, if not return Empty result
+      // While it may seem we don't tune the lookback of the leaf raw queries to exactly what we need from each
+      // partition, in reality it doesnt matter as despite a longer lookback, the actual data exported will be at most
+      // what that partition contains.
+      val (startTime, endTime) = (qParams.startSecs, qParams.endSecs)
+      val totalExpectedRawExport = (endTime - startTime) + lookbackMs + offsetMs
+      if (queryConfig.supportRemoteRawExport &&
+        queryConfig.maxRemoteRawExportTimeRange.toMillis > totalExpectedRawExport) {
+        val newLp = rewritePlanWithRemoteRawExport(logicalPlan, IntervalSelector(startTime * 1000, endTime * 1000))
+        walkLogicalPlanTree(newLp, qContext, true).plans
+      } else {
+        if (queryConfig.supportRemoteRawExport) {
+          logger.warn(
+            s"Remote raw export is supported and the $totalExpectedRawExport ms" +
+              s" is greater than the max allowed raw export duration of ${queryConfig.maxRemoteRawExportTimeRange}")
+        } else {
+          logger.warn("Remote raw export not enabled")
+        }
+        Seq(EmptyResultExec(qContext, dataset.ref, inProcessPlanDispatcher))
```

Comment:
The older implementation would actually no return results if the lookback goes across partitions.  The reason I'm return empty result along with a warn message  is because failures will show up on user side( which doesn't happen today when multiple partitions are involved). I guess I can improve the log message by logging the query.